pump/storage/storage.go

```diff
@@ -686,19 +685,52 @@ func (a *Append) doGCTS(ts int64) {
 		wg.Done()
 	}()
 
-	batch := new(leveldb.Batch)
 	l0Trigger := defaultStorageKVConfig.CompactionL0Trigger
 	if a.options.KVConfig != nil && a.options.KVConfig.CompactionL0Trigger > 0 {
 		l0Trigger = a.options.KVConfig.CompactionL0Trigger
 	}
+	var alreadyGcTS int64
+	for {
+		alreadyGcTS = a.batchGC(alreadyGcTS, ts, 100, l0Trigger, func() (int, error) {
+			nStr, err := a.metadata.GetProperty("leveldb.num-files-at-level0")
+			if err != nil {
+				return 0, errors.Annotate(err, "get `leveldb.num-files-at-level0` property failed")
+			}
 
+			l0Num, err := strconv.Atoi(nStr)
+			if err != nil {
+				return 0, errors.Annotatef(err, "parse `leveldb.num-files-at-level0` result to int failed, str: %s", nStr)
+			}
+			return l0Num, nil
+		})
```

Comment:
Assign this function to a variable outside the loop and use the variable here would make the code clearer?